### ISSUE TITLE
Incompatible with NivoSlider

### ISSUE THREAD

**Issue by mstummer** [bfred-it/object-fit-images#57]
Hey there!

First of all: thanks heaps for this lib, saves me days of headaches!

It works 99% on my website, except that it seems to remove the 'src' attribute from images without 'object-fit', at least in a particular case. I use a plugin called MetaSlider (www.metaslider.com) on my Wordpress website, and that stops working on Edge etc when using your lib. I've inspected the element, and as far as I can see there is no 'object-fit' in there (see screengrabs below), so in theory it should be unaffected. Other images on my page do stay untouched, so there might be an edge case here somehow.

I'm good for now, will just use objectFitImages() on pages where I actually need it, instead of globally in the header. Just thought I'd let you know, maybe you spot something in the screengrabs...

Cheers,
Max

```
<head>
	...
	<!--
	object-fit browser compatibility
	https://github.com/bfred-it/object-fit-images
	-->
	<script src="https://unpkg.com/object-fit-images@2.5.9/dist/ofi.browser.js"></script>
</head>

<body>
<script>
    jQuery(function () { objectFitImages() });
</script>
...
```
Working, without object-fit-images:
![image](https://cloud.githubusercontent.com/assets/16602214/21584757/aaadded2-d114-11e6-863b-274ba538cd9b.png)

Src removed, with object-fit-images:
![image](https://cloud.githubusercontent.com/assets/16602214/21584743/48b60556-d114-11e6-818c-0cecbc2392ac.png)

Inspect img:
![image](https://cloud.githubusercontent.com/assets/16602214/21584605/634c6b3c-d112-11e6-9d3d-20688dc3b1ef.png)
...cont...
![image](https://cloud.githubusercontent.com/assets/16602214/21584607/810cccac-d112-11e6-99cc-80d7c26a501e.png)

**Comment by fregante** [comment]
That's strange, I don't unset the src anywhere. I think that there's some conflicting script that's removing it.

Can you run this in the console and tell me what you see?

```js
document.querySelector('.nivo-main-image')['data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==']
```

If it's `undefined` OFI isn't running on that image. Otherwise you should see an object.

Also, can you try just including OFI without calling `objectFitImages()`? Does that work?

**Comment by fregante** [comment]
Alternatively, try this new beta version: https://unpkg.com/object-fit-images@3.0.0-0/dist/ofi.browser.js

And this is the new test code:

```js
document.querySelector('.nivo-main-image')['bfred-it:object-fit-images']
```

**Comment by mstummer** [comment]
Hey Frederico!

Thanks for looking into this! So, I had another play with it... Testing in Microsoft Edge 38.14393.0.0

- add OFI, don't call objectFitImages() >> everything works fine
- add OFI, call objectFitImages() >> NivoSlider is impty, src not set
- add OFI 3.0.0-0, call objectFitImages() >> NivoSlider is impty, src not set, error "invalid calling object, ofi.browser.js (2,2179)"
- "document.querySelector(...." >> always returns 'undefined'.

Having said all that, I had a brief look at MetaSlider/NivoSlider. It looks like when the page loads, src is not set on nivo-main-image, and the slider has 0 height. When their js code executes, it injects the different src filepaths, and the images pop into place. So it looks like you're right, OFI isn't doing anything on that image... but it somehow interferes with the MetaSlider code and keeps it from setting the src path.

Not sure if you can/wanna do much about that... I got it working for me by putting all the object-fits into different classes, and using those on the img's... then using a selector on OFI instead of running it on everything. Works fine for me.

```
CSS
.object-fit-cover {
    object-fit: cover;
    font-family: 'object-fit: cover;';
}
.object-fit-fill {
    object-fit: fill;
    font-family: 'object-fit: fill;';
}

HTML
<img class="object-fit-cover" src="...">

JS
objectFitImages(jQuery( "img[class*='object-fit-']" ));
```

**Comment by mstummer** [comment]
I have to admit that I usually don't do much web, so no idea how this all works. Since your code just runs once and then applies to all loaded images later ("auto-mode" as you described it), I'm guessing it uses some sort of hook? IMG creation event? I'm just guessing... (I'm usually a java dev)... if there was such an event, and you were using it, sounds a bit like NivoSlider might be using the same event, but your code gets executed first and "eats" the event. That's me just guessing... :)

**Comment by fregante** [comment]
Can you point me to the NivoSlider you're using? I see a few versions.

Also is the site online somewhere? I could test it there directly.

**Comment by mstummer** [comment]
This is what I'm using: MetaSlider (https://wordpress.org/plugins/ml-slider/)... that has a few different modes that you can choose from, and one of them is NivoSlider.
You can try this one to test: www.trailraces.co.nz/home/

**Comment by innocentiv** [comment]
Error "invalid calling object, ofi.browser.js (2,2179)" is fixed in https://github.com/bfred-it/object-fit-images/commit/f51f85449b842d90921522cce45c953692433bf8

**Comment by fregante** [comment]
@mstummer: that fix has been published in 3.0.1, can you test the new version?
